File: cfacq/app/domaine/media.ts
import {Fichier} from './fichier';

export class Media {
  id: number;
  image: Fichier;
  description: string;
  link: string;
  date: string;
  dateMedia: Date;
  dateMediaString: string;
  traite: boolean;

  constructor(image?: Fichier, description?: string, link?: string, date?: string, dateMedia?: Date, dateMediaString?: string, traite?: boolean ) {
    this.image = image;
    this.description = description;
    this.link = link;
    this.date = date;
    this.dateMedia = dateMedia;
    this.dateMediaString = dateMediaString;
    this.traite = traite;
  }
}

File: cfacq/backend/domaine/Media.java
package sitecfacq.domaine;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.voodoodyne.jackson.jsog.JSOGGenerator;
import sitecfacq.domaine.common.Fichier;
import sitecfacq.domaine.common.GescoEntity;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToOne;
import java.util.Date;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
@Entity
@JsonIdentityInfo(generator=JSOGGenerator.class)
public class Media extends GescoEntity {
  @OneToOne(cascade = CascadeType.ALL)
  private Fichier image;
  private String description;
  private String link;
  private Date date;

  public Fichier getImage() {
    return image;
  }

  public void setImage(Fichier image) {
    this.image = image;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getLink() {
    return link;
  }

  public void setLink(String link) {
    this.link = link;
  }

  public Date getDate() {
    return date;
  }

  public void setDate(Date date) {
    this.date = date;
  }
}
